src/plat/darwin/detect.c

```diff
@@ -40,24 +45,22 @@
 */
 void detect_distro(void)
 {
-#if __MAC_OS_X_VERSION_MIN_REQUIRED <= 1070
-	int major, minor, bugfix;
-
-	Gestalt(gestaltSystemVersionMajor, (SInt32 *) &major);
-	Gestalt(gestaltSystemVersionMinor, (SInt32 *) &minor);
-	Gestalt(gestaltSystemVersionBugFix, (SInt32 *) &bugfix);
-
-	snprintf(distro_str, MAX_STRLEN, "Max OS X %d.%d.%d", major, minor, bugfix);
-#else
-	FILE *distro_file;
-	char distro_vers_str[MAX_STRLEN] = {0};
-
-	distro_file = popen("sw_vers -productVersion | tr -d '\\n'", "r");
-	fgets(distro_vers_str, MAX_STRLEN, distro_file);
-	pclose(distro_file);
-
-	snprintf(distro_str, MAX_STRLEN, "Mac OS X %s", distro_vers_str);
-#endif
+	char *codenames[] = {"Cheetah", "Puma", "Jaguar", "Panther", "Tiger", "Leopard", "Snow Leopard", "Lion", "Mountain Lion", "Mavericks", "Yosemite", "El Capitan", "Sierra", "High Sierra", "Mojave"};
+	CFArrayRef split = CFStringCreateArrayBySeparatingStrings(NULL, CFPreferencesCopyAppValue(CFSTR("ProductVersion"), CFSTR("/System/Library/CoreServices/SystemVersion")), CFSTR("."));
+	unsigned maj = CFStringGetIntValue(CFArrayGetValueAtIndex(split, 0));
+	unsigned min = CFStringGetIntValue(CFArrayGetValueAtIndex(split, 1));
+	unsigned fix = 0;
+	if(CFArrayGetCount(split) == 3)
```

Comment:
Style nit (here are throughout the changes): Put a space after the `if` token and always use braces on branches.